lib/coach/handler.rb

```diff
@@ -25,16 +25,28 @@ def call(env)
       start = Time.now
 
       publish('coach.handler.start', start_event.dup)
-      response = chain.instrument.call
 
-      finish = Time.now
-      publish('coach.handler.finish',
-              start, finish, nil,
-              start_event.
-                merge(response: { status: response[0] },
-                      metadata: context.fetch(:_metadata, {})))
+      begin
+        response = chain.instrument.call
+      rescue => e
+        finish = Time.now
+        publish('coach.handler.error',
+                start, finish, nil,
+                start_event.merge(
+                  response: { status: 500 },
```

Comment:
So if you look at the literature around monitoring latency, it's a known thing that you should separate your errors from successful requests [1](https://blog.netsil.com/the-4-golden-signals-of-api-health-and-performance-in-cloud-native-applications-a6e87526e74) because en error quite frequently processes faster than a valid request. So I think separating this out at the coach level is probably a non-obvious way of doing things, whereas sampling at your data aggregation tool for non-`5XX` response latency is likely a better solution.

Also it's not guessing a 500 right? When we get to this point, we've encountered a server error. It's very unusual that we would turn a 500 into anything that isn't a 500 upstream in our Rack middleware, and if we do then we'd be able to see that in our outer layers of monitoring (edge lbs, etc) and we'd probably want to know the distinction.